gk/fib.c

```diff
@@ -957,15 +957,17 @@ ether_cache_put(struct gk_fib *neigh_fib,
 	int ret, ref_cnt;
 	struct gk_fib *neighbor_fib = neigh_fib;
 
-	ref_cnt = rte_atomic32_sub_return(&eth_cache->ref_cnt, 1);
-	if (ref_cnt < 0) {
+	while ((ref_cnt = rte_atomic32_read(&eth_cache->ref_cnt)) >= 2) {
+		if (likely(rte_atomic32_cmpset((volatile uint32_t *)
+				&eth_cache->ref_cnt.cnt,
```

Comment:
The message is: ``error: pointer targets in passing argument 1 of ‘rte_atomic32_cmpset’ differ in signedness [-Werror=pointer-sign]``, because ``&eth_cache->ref_cnt.cnt`` has type ``volatile int32_t``.

Reply:
Thank you. It seems to be a "type" bug in DPDK. Let's leave it as it is for now.